### PR TITLE
Sleepers now use power cells [PAID CODE]

### PR DESCRIPTION
**Pull request by Iamgoofball**
ported from https://github.com/WaspStation/WaspStation-1.0/pull/60 at the request of the contractor, forgot to include this originally lmao my bad

## About The Pull Request

Sleepers now use power cells to dispense chemicals.
dreamseeker_HSJxfsopch

## Why It's Good For The Game

I was paid in sprites to do this.

## Changelog
:cl:
balance: Sleepers now use power cells to dispense chemicals.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
bruh

**Comment by ike709** [comment]
Conflicts, needs to conform to TGUI3's sleeper UI. See #1623

File: code/game/machinery/Sleeper.dm [hunk 1]
```diff
@@ -12,6 +12,10 @@
 	density = FALSE
 	state_open = TRUE
 	circuit = /obj/item/circuitboard/machine/sleeper
+	ui_x = 250
+	ui_y = 550
+	var/obj/item/stock_parts/cell/cell //The sleeper's power cell.
+	var/poweruse = 250 //How much power it uses per injection, minimum 10 charge/10u
```
Comment (Superyodeler) [inline code/game/machinery/Sleeper.dm]: ui_x is changed, but 16-18 are taken directly from wasp code
You didn't even change the comments

File: code/game/machinery/Sleeper.dm [hunk 2]
```diff
@@ -43,11 +49,15 @@
 
 	efficiency = initial(efficiency)* E
 	min_health = initial(min_health) * E
+	poweruse = max(100, 300/efficiency)
```
Comment (Superyodeler) [inline code/game/machinery/Sleeper.dm]: Same values, same place

File: code/game/machinery/Sleeper.dm [hunk 3]
```diff
@@ -159,6 +169,11 @@
 
 /obj/machinery/sleeper/ui_data()
 	var/list/data = list()
+	data["cell"] = list()
+	if(cell) // someone who knows how tgui works please make this work
+		data["cell"]["poweruse"] = poweruse
+		data["cell"]["maxCharge"] = cell.maxcharge
+		data["cell"]["charge"] = cell.charge ? cell.charge : "0"
```
Comment (Superyodeler) [inline code/game/machinery/Sleeper.dm]: Copy pasted from Wasp with an added comment

**Comment by Iamgoofball** [comment]
This PR does not touch the UI.

File: code/game/machinery/Sleeper.dm [hunk 4]
```diff
@@ -12,6 +12,10 @@
 	density = FALSE
 	state_open = TRUE
 	circuit = /obj/item/circuitboard/machine/sleeper
+	ui_x = 250
+	ui_y = 550
```
Comment (ike709) [inline code/game/machinery/Sleeper.dm]: 1

File: code/game/machinery/Sleeper.dm (hunk 3 above)
Comment (ike709) [inline code/game/machinery/Sleeper.dm]: 2

**Comment by ike709** [comment]
> This PR does not touch the UI.

🤔

File: code/game/machinery/Sleeper.dm [hunk 5]
```diff
@@ -227,11 +242,18 @@
 	to_chat(user, "<span class='warning'>You scramble the sleeper's user interface!</span>")
 
 /obj/machinery/sleeper/proc/inject_chem(chem, mob/user)
-	if((chem in available_chems) && chem_allowed(chem))
+	if((chem in available_chems) && chem_allowed(chem) && cell.charge > poweruse)
+		cell.use(poweruse)
```
Comment (Superyodeler) [inline code/game/machinery/Sleeper.dm]: Copy pasted from wasp

File: code/game/machinery/computer/cloning.dm [hunk 6]
```diff
@@ -23,6 +23,7 @@
 
 	var/loading = FALSE // Nice loading text
 	var/autoprocess = FALSE
+	var/obj/item/reagent_containers/glass/beaker = null //Beaker full of what SHOULD be synthflesh
```
Comment (Superyodeler) [inline code/game/machinery/computer/cloning.dm:26]: Taken from another [wasp PR](https://github.com/WaspStation/WaspStation-1.0/pull/60).

File: code/game/objects/items/circuitboards/machine_circuitboards.dm [hunk 7]
```diff
@@ -5,7 +5,9 @@
 		/obj/item/stock_parts/matter_bin = 1,
 		/obj/item/stock_parts/manipulator = 1,
 		/obj/item/stack/cable_coil = 1,
-		/obj/item/stack/sheet/glass = 2)
+		/obj/item/stack/sheet/glass = 2,
+		/obj/item/stock_parts/cell = 1)
+	def_components = list(/obj/item/stock_parts/cell = /obj/item/stock_parts/cell/high)
```
Comment (Superyodeler) [inline code/game/objects/items/circuitboards/machine_circuitboards.dm:10]: Copy pasted from wasp

**Review by Superyodeler — COMMENTED** [review]
Uh oh, plagiarism!
Most of this pr is taken from [WaspStation Code](https://github.com/WaspStation/WaspStation-1.0); [This PR](https://github.com/WaspStation/WaspStation-1.0/pull/53) to be specific.

**Comment by Iamgoofball** [comment]
@Superyodeler i was asked to port that PR, forgot to put it in the OP, thanks for linking it

**Comment by ike709** [comment]
> plagiarism!
> AGPL code

lol

**Comment by Iamgoofball** [comment]
we in the business call this a "port"

**Comment by Superyodeler** [comment]
As long as you give credit, I'm happy

**Comment by Iamgoofball** [comment]
this bounty has been redirected, closing this PR with no survivors

**Pull request closed without merging**